cmd/soroban-cli/src/commands/tx/new/revoke_sponsorship.rs

```diff
@@ -0,0 +1,179 @@
+use clap::Parser;
+use soroban_sdk::xdr;
+
+use super::clawback_claimable_balance::parse_balance_id;
+use crate::{commands::tx, config::address, tx::builder};
+
+#[derive(Parser, Debug, Clone)]
+#[group(skip)]
+pub struct Cmd {
+    #[command(flatten)]
+    pub tx: tx::args::Args,
+
+    #[command(flatten)]
+    pub op: Args,
+}
+
+#[derive(Debug, clap::Args, Clone)]
+pub struct Args {
+    /// Account ID (required for all sponsorship types)
+    #[arg(long)]
+    pub account_id: address::UnresolvedMuxedAccount,
+
+    /// Asset for trustline sponsorship (format: CODE:ISSUER)
+    #[arg(long, group = "sponsorship_type")]
+    pub asset: Option<builder::Asset>,
+
+    /// Data name for data entry sponsorship
+    #[arg(long, group = "sponsorship_type")]
+    pub data_name: Option<String>,
+
+    /// Offer ID for offer sponsorship
+    #[arg(long, group = "sponsorship_type")]
+    pub offer_id: Option<u64>,
+
+    /// Pool ID for liquidity pool sponsorship. Accepts multiple formats:
+    /// - API format with type prefix (72 chars): 000000006f2179b31311fa8064760b48942c8e166702ba0b8fbe7358c4fd570421840461
+    /// - Direct hash format (64 chars): 6f2179b31311fa8064760b48942c8e166702ba0b8fbe7358c4fd570421840461
+    /// - StrKey format (base32): LAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
```

Comment:
I think we should refer to this as "address" instead of strkey. Strkey is the standard, but for contract developers the string format is known as addresses.
```suggestion
    /// - Address format (base32): LAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
```